br/pkg/lightning/backend/external/byte_reader.go

```diff
@@ -93,6 +93,10 @@ func (r *byteReader) readNBytes(n int) (*[]byte, error) {
 	return &auxBuf, nil
 }
 
+func (r *byteReader) reset() {
```

Comment:
not important: we can also set elements to nil before truncate the `retPointers`

Reply:
maybe also add more reset() in unit tests